pkg/observ/observ.go

```diff
@@ -17,16 +18,56 @@ type observabilityContext struct {
 	spanCtx context.Context
 }
 
-// RegisterTraceExporter returns a jaeger exporter for exporting traces to opencensus.
+// RegisterExporter determines the type of TraceExporter service for exporting traces from opencensus
+// User can choose from multiple tracing services (datadog, jaegar)
+// RegisterExporter returns the 'Flush' function for that particular tracing service
+// TODO: Extend beyond jaeger and datadog
+func RegisterExporter(TraceExporter, URL, service, ENV string) (func(), error) {
+
+	const op errors.Op = "RegisterExporter"
+
+	switch TraceExporter {
+	case "jaeger":
+		ex, err := registerJaegerExporter(URL, service, ENV)
+		return jaegerFlush(ex), errors.E(op, err)
```

Comment:
`return ex.Flush, nil` -- no need to create a function that returns a function that calls a function :)